float32[] joint_positions
---
float32[] endpoint_pose